pkg/promlib/querydata/response_test.go

```diff
@@ -61,10 +62,47 @@ func TestAddMetadataToMultiFrame(t *testing.T) {
 	t.Run("when you have native histogram result", func(t *testing.T) {
 		qd := QueryData{exemplarSampler: exemplar.NewStandardDeviationSampler}
 		resBody := `{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"__name__":"rpc_durations_native_histogram_seconds","instance":"nativehisto:8080","job":"prometheus"},"histograms":[[1729529685,{"count":"7243102","sum":"72460202.93145595","buckets":[[0,"1.8340080864093422","2","10"],[0,"2","2.1810154653305154","68"]]}],[1729529700,{"count":"7243490","sum":"72464056.03309634","buckets":[[0,"1.8340080864093422","2","10"],[0,"2","2.1810154653305154","68"]]}],[1729529715,{"count":"7243880","sum":"72467935.35871512","buckets":[[0,"1.8340080864093422","2","10"],[0,"2","2.1810154653305154","68"]]}]]}]}}`
-		res := &http.Response{Body: io.NopCloser(bytes.NewBufferString(resBody))}
+		res := &http.Response{Body: io.NopCloser(bytes.NewBufferString(resBody)), StatusCode: 200}
 		result := qd.parseResponse(context.Background(), &models.Query{}, res)
 		assert.Nil(t, result.Error)
 		assert.Len(t, result.Frames, 1)
 		assert.Equal(t, "yMin", result.Frames[0].Fields[1].Name)
 	})
 }
+
+// Helper function to create mock HTTP response
+func createMockResponse(statusCode int, body string) *http.Response {
+	return &http.Response{
+		StatusCode: statusCode,
+		Body:       io.NopCloser(bytes.NewReader([]byte(body))),
+	}
+}
+
+func TestParseResponse_ErrorCases(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name       string
+		statusCode int
+		body       string
+	}{
+		{"500 Internal Server Error", http.StatusInternalServerError, `{"error":"internal server error"}`},
+		{"404 Not Found", http.StatusNotFound, `{"error":"not found"}`},
+		{"401 Unauthorized", http.StatusUnauthorized, `{"error":"unauthorized"}`},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res := createMockResponse(tt.statusCode, tt.body)
+			q := &models.Query{}
+			qd := QueryData{exemplarSampler: exemplar.NewStandardDeviationSampler}
+			qd.log = log.New()
+			resp := qd.parseResponse(ctx, q, res)
+
+			assert.NotNil(t, resp.Error)
```

Comment:
`require` is more suitable here, since you don't want to continue the test if `resp.Error` is nil:
```suggestion
			require.Error(t, resp.Error)
```